### PR TITLE
Refactor api request errors validation

### PR DESCRIPTION
**Pull request by alexsds**
@permiakov

### CODE REVIEW

File: src/EloquaProvider.php [hunk 1]
```diff
@@ -62,7 +63,9 @@ protected function getDefaultHeaders()
      */
     protected function checkResponse(ResponseInterface $response, $data)
```
Comment (permiakov) [inline src/EloquaProvider.php:63]: Why?

Comment (alexsds) [reply src/EloquaProvider.php:63]: Eloqua will respond with status codes in that diapason if some errors in request params ([Validation-errors](http://docs.oracle.com/cloud/latest/marketingcs_gs/OMCAB/index.html#Developers/GettingStarted/API%20requests/Validation-errors.htm))

Comment (permiakov) [reply src/EloquaProvider.php:63]: is $data['error'] always present?

Comment (alexsds) [reply src/EloquaProvider.php:63]: no

**Comment by permiakov** [comment]
Please add tests

**Comment by alexsds** [comment]
added

**Pull request merged**